tests/core_tests.py

```diff
@@ -404,6 +405,219 @@ def test_only_owners_can_save(self):
+                slug='{}_imported'.format(title.lower()),
+            )
+
+        def find_dash(dash_id):
+            return session.query(models.Dashboard).filter_by(
+                id=dash_id).first()
+
+        def get_slc(name, table, slc_id):
+            return models.Slice(
+                id=slc_id,
+                slice_name=name,
+                datasource_type='table',
+                perm='[main].[{}](id:{})'.format(table, slc_id)
+            )
+
+        # Case 1. Import empty dashboard
+        empty_dash = get_dashboard('empty_dashboard', id=1001)
+        imported_dash_id_1 = models.Dashboard.import_dashboard(
+            empty_dash, import_time=1989)
+        # Dashboard import fails if there are no slices
+        self.assertIsNone(find_dash(imported_dash_id_1))
+
+        # Case 2. Import dashboard with 1 new slice.
+        dash_with_1_slice = get_dashboard(
+            'dash_with_1_slice',
+            slcs=[get_slc('health_slc', 'wb_health_population', 1001)],
+            id=1002)
+        imported_dash_id_2 = models.Dashboard.import_dashboard(
+            dash_with_1_slice, import_time=1990)
+        imported_dash_2 = find_dash(imported_dash_id_2)
+        self.assertEquals('dash_with_1_slice', imported_dash_2.dashboard_title)
+        self.assertEquals(1, len(imported_dash_2.slices))
+        self.assertEquals('wb_health_population',
+                          imported_dash_2.slices[0].datasource.table_name)
+        self.assertEquals(
+            {"remote_id": 1002, "import_time": 1990},
+            json.loads(imported_dash_2.json_metadata))
+
+        # Case 3. Import dashboard with 2 new slices.
+        dash_with_2_slices = get_dashboard(
+            'dash_with_2_slices',
+            slcs=[get_slc('energy_slc', 'energy_usage', 1002),
+                  get_slc('birth_slc', 'birth_names', 1003)],
+            id=1003)
+        imported_dash_id_3 = models.Dashboard.import_dashboard(
+            dash_with_2_slices, import_time=1991)
+        imported_dash_3 = find_dash(imported_dash_id_3)
+        self.assertEquals('dash_with_2_slices', imported_dash_3.dashboard_title)
+        self.assertEquals(2, len(imported_dash_3.slices))
+        self.assertEquals(
+            {"remote_id": 1003, "import_time": 1991},
+            json.loads(imported_dash_3.json_metadata))
+        imported_dash_slice_ids_3 = {s.id for s in imported_dash_3.slices}
+
+        # Case 4. Override the dashboard and 2 slices
+        dash_with_2_slices_new = get_dashboard(
+            'dash_with_2_slices_new',
+            slcs=[get_slc('energy_slc', 'energy_usage', 1002),
+                  get_slc('birth_slc', 'birth_names', 1003)],
+            id=1003)
+        imported_dash_id_4 = models.Dashboard.import_dashboard(
+            dash_with_2_slices_new, import_time=1992)
+        imported_dash_4 = find_dash(imported_dash_id_4)
+        self.assertEquals('dash_with_2_slices_new',
```

Comment:
`assert imported_dash_4.dashboard_title == imported_dash_2.dashboard_title` instead of hard coding the title twice